src/contracts/pods/EigenPodManager.sol

```diff
@@ -314,12 +330,39 @@ contract EigenPodManager is
     }
 
     /// @notice Returns the Beacon block root at `timestamp`. Reverts if the Beacon block root at `timestamp` has not yet been finalized.
-    function getBlockRootAtTimestamp(uint64 timestamp) external view returns (bytes32) {
+    function getBlockRootAtTimestamp(uint64 timestamp) public view returns (bytes32) {
         bytes32 stateRoot = beaconChainOracle.timestampToBlockRoot(timestamp);
         require(
             stateRoot != bytes32(0),
             "EigenPodManager.getBlockRootAtTimestamp: state root at timestamp not yet finalized"
         );
         return stateRoot;
     }
+
+    //make a request for a proof to the succinct gateway
+    function requestProofViaSuccinctGateway(
```

Comment:
I'm confused--is this the entrypoint for the user requesting a withdrawal? If that's the case, then they shouldn't be able to pass in a FUNCTION_ID, it should be stored and we should just use that stored version (also the stored FUNCTION_ID should have upgradeability similar to the gateway)